engines/python/setup/djl_python/rolling_batch/vllm_rolling_batch.py

```diff
@@ -97,11 +97,78 @@ def translate_to_engine_params(self, parameters: dict) -> dict:
             parameters["ignore_eos"] = parameters.pop("ignore_eos")
         return parameters
 
-    def get_request_id(self, request):
+    @stop_on_any_exception
+    def inference(self, input_data: list[str], parameters: list[dict]) -> list:
         """
-        Get request id that will be set to backend engine request
+        Adds new requests and gets output tokens from the backend.
+
+        :param input_data: List of input prompts.
+        :param parameters: List of settings pertaining to each request.
+
+        :return results: List of dictionaries, one for each request, that contain output tokens and other data.
         """
-        return random_uuid()
+        batch_size = len(input_data)
+        new_requests = self.get_new_requests(input_data, parameters,
+                                             batch_size)
+        # step 0: register new requests to engine
+        for request in new_requests:
+            request_id = random_uuid()
+            params = self.translate_vllm_params(request.parameters)
+            sampling_params = SamplingParams(**params)
+            self.engine.add_request(request_id, request.input_text,
+                                    sampling_params)
+            self.request_cache[request_id] = {
+                "curr_length": 0,
+                "text": "",
+                "cumulative_logprob": 0.0,
+                "log_prob": 0.0,
+                "finished": False,
+                "finish_reason": None
+            }
+        request_outputs = self.engine.step()
+
+        # step 1: put result to cache
+        for request_output in request_outputs:
+            self.request_cache = update_request_cache_with_output(
```

Comment:
moved the earlier logic here and and speculative decoding metrics logging to `rolling_batch_vllm_utils.py` script